src/log/zipkin.cpp

```diff
@@ -14,6 +14,9 @@
 
 namespace fc {
 
+// signal safe, thread atomic and lock free
+std::atomic_flag do_sighup = ATOMIC_FLAG_INIT;
```

Comment:
Why is this file scope instead of in `zipkin::impl` ?

Reply:
Static method `handle_sighup()` defined in class `zipkin_config` needs to access this var and update it when a SIGHUP signal is received.  It's not straight forward to call this var if it's declared in `zipkin::impl`. Is there any side effect to put it in this file scope? I would change it if there is.